roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -33,6 +33,14 @@ def run(self, tmp, task_vars):
         configured_min = int(get_var(task_vars, "openshift_check_min_host_disk_gb", default=0)) * 10**9
         min_free_bytes = configured_min or recommended_min
 
+        # if an "upgrade" context is set, lower the minimum disk requirement
+        # as this signifies an in-place upgrade - the node might have the
+        # required total disk space, but some of that space may already be
+        # in use by the existing OpenShift deployment.
+        context = get_var(task_vars, "r_openshift_health_checker_playbook_context", default="")
+        if context == "upgrade":
+            min_free_bytes /= 2
```

Comment:
Not able to find a definite requirement in the docs for disk size during an upgrade.

> Or, this could be re-purposed as a "health" check when it's not in a fresh install context.

That could also work. We could fail the check in this context if disk usage is something like > 90%